### ISSUE TITLE
Performance issues with version 2.5

### ISSUE THREAD

**Issue by danielpes** [cytoscape/cytoscape.js#1200]
Hi,

We are noticing some performance issues after upgrading our project from cytoscape.js version 2.4 to 2.5. Our graphs do not have many edges (around 50), but we expect heavy interaction (pan/zoom and nodes movement). 

As it is difficult to explain our context, we did some experiments with the ["Performance Tuning"](http://js.cytoscape.org/demos/57e7cc43342193d9d21a/) example present in cytoscape.js homepage and we were able to reproduce the problem. The following image shows CPU usage with cytoscape.js version 2.4.6:

![2-4-6_2](https://cloud.githubusercontent.com/assets/1709357/11874289/fc7b615c-a4c5-11e5-926b-c5173d49f95d.png)

The red arrow shows the period **during** heavy interaction (pan/zoon), while the blue arrow shows the period **after** interaction. The following image shows the CPU usage level after interaction, still for version 2.4.6:

![2-4-6](https://cloud.githubusercontent.com/assets/1709357/11874345/41ff42b6-a4c6-11e5-8d60-f1bfe64d37b6.png)

Now, for cytoscape.js version 2.5.2, we had the following results:

![2-5-2_2](https://cloud.githubusercontent.com/assets/1709357/11874354/50efec26-a4c6-11e5-89ec-f31e2b8d5389.png)

Again, the arrows have the same meaning as before. Note how the CPU level does not drop as much as before after interaction. The following image shows a CPU usage of 29% after stopping the interaction:

![2-5-2](https://cloud.githubusercontent.com/assets/1709357/11874362/5ab771c0-a4c6-11e5-9322-8a05d0c8479e.png)

In our project (we use cytoscape.js in an angular.js directive), many interactions cause updates in the elements data and lots of redrawing, so for us the CPU keeps above 80% even after interaction. We tested with cytoscape.js 2.4.6 and this problem does not occur.

The performance tests with the example from the website were made in Chrome version **44.0.2403.130** 64-bit (47 too)  in a **Linux Mint 17.1 64-bit**.

**Comment by maxkfranz** [comment]
Regarding rendering, 2.5 has changes to rely more on the browser's framerate limiting rather than our own.  This increases (approximately doubles) the framerate on certain setups.  It looks like you're rendering about double the frames per second, thereby doubling the CPU usage as expected.

There is no difference in actual performance: The graph is responsive either way.  The only difference is that 2.5 will give smoother rendering for more setups.

**Comment by danielpes** [comment]
Hi Max, thank you for your response. 

We noticed the doubling of CPU usage during interactions, however, the main problem for us is that the usage is not lowering enough **after** we stop all interactions. In the case shown here, the CPU usage level did not get below 25%, and in our project it keeps above 80%. Only if we pause script execution using F8 in Chrome Developer Tools the CPU the usage goes back to normal.

**Comment by danielpes** [comment]
Hi,

We managed to solve the problem in our project by using startBatch() and endBatch() around some loops that update all elements data. We noticed that these loops were being called frequently and causing too many redraws. Now the CPU usage behavior in our project is similar to the "Performance Tuning" example and we considerer it good for our use.

Thanks for the support and sorry for any inconvenience.
